internal/provider/resource_model.go

```diff
@@ -486,9 +497,14 @@ func (r *modelResource) Delete(ctx context.Context, req resource.DeleteRequest,
 		return
 	}
 
-	err := r.client.Models.DestroyModel(juju.DestroyModelInput{
-		UUID: state.ID.ValueString(),
-	})
+	modelUUID := state.UUID.ValueString()
+	if modelUUID == "" {
```

Comment:
Is this possible? Shouldn't we always have a model's UUID at the time of a destroy operation?

Reply:
We should always have the UUID, but I'm not completely sure in the case where the model was created with an older version of the terraform provider. In that case, what is Read is not called before Delete?  Erring on the side of caution.